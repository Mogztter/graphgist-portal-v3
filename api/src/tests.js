import { ApolloServer } from "apollo-server-express";
import { createTestClient as apolloCreateTestClient } from "apollo-server-testing";
import neo4j from 'neo4j-driver';
import { makeAugmentedSchema } from "neo4j-graphql-js";
import dotenv from "dotenv";
import { typeDefs } from "./graphql-schema";

dotenv.config();

const schema = makeAugmentedSchema({
  typeDefs,
  config: {
    mutation: false
  }
});

const driver = neo4j.driver(
  process.env.NEO4J_URI || "bolt://localhost:7687",
  neo4j.auth.basic(
    process.env.NEO4J_USER || "neo4j",
    process.env.NEO4J_PASSWORD || "neo4j"
  )
);

const server = new ApolloServer({
  context: { driver },
  schema: schema,
  mocks: true
});

export function createTestClient() {
  return apolloCreateTestClient(server);
}
