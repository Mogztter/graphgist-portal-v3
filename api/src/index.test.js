import { createTestClient } from "./tests.js";
import gql from "graphql-tag";

describe("graphql endpoint", () => {
  const { query, mutate } = createTestClient();

  test("can query graphgist", async () => {
    const res = await query({
      query: gql`
        {
          GraphGist(first: 10) {
            title
          }
        }
      `,
    });
    expect(res).toMatchSnapshot();
  });

  test("can preview graphgist", async () => {
    const res = await mutate({
      mutation: gql`
        mutation Preview($asciidoc: String!) {
          PreviewGraphGist(asciidoc: $asciidoc)
        }
      `,
      variables: { asciidoc: "== Testtt" },
    });
    expect(res).toMatchSnapshot();
  });
});
