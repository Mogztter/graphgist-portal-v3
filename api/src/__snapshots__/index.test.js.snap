// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`graphql endpoint can query graphgist 1`] = `
Object {
  "data": Object {
    "GraphGist": Array [
      Object {
        "title": "Hello World",
      },
      Object {
        "title": "Hello World",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
